server/norace_test.go

```diff
@@ -9599,3 +9599,58 @@ func TestNoRaceJetStreamKVReplaceWithServerRestart(t *testing.T) {
 		t.Fatalf("Encountered errors")
 	}
 }
+
+func TestNoRaceMemStoreCompactDuration(t *testing.T) {
+	//Load MemStore so that it is full
+	subj, msg := "foo", make([]byte, 1000)
+	storedMsgSize := memStoreMsgSize(subj, nil, msg)
+
+	toStore := uint64(500_000)
+	maxBytes := storedMsgSize * toStore
+
+	ms, err := newMemStore(&StreamConfig{Storage: MemoryStorage, MaxBytes: int64(maxBytes)})
+	require_NoError(t, err)
+	defer ms.Stop()
+
+	for i := uint64(0); i < toStore; i++ {
+		ms.StoreMsg(subj, nil, msg)
+	}
+	state := ms.State()
+	if state.Msgs != toStore {
+		t.Fatalf("Expected %d msgs, got %d", toStore, state.Msgs)
+	}
+	if state.Bytes != storedMsgSize*toStore {
+		t.Fatalf("Expected bytes to be %d, got %d", storedMsgSize*toStore, state.Bytes)
+	}
+
+	toStoreOnTop := uint64(10_000_000)
+
+	//1st run: Load additional 10 Mio messages then compact
```

Comment:
Mio?

Reply:
I will think about a lower numer. Maybe be just removing N-1 as initially stated by you. 
I just wanna make sure that this test never breaks as timing related tests have the tendency to do so if the build server is slightly slower as someone is doing an update etc... I.e., I want to make sure not to hand over a test that breaks randomly every now and then :-) 
With 1 mio entries I ran into false positives on my local machine already... 
With purgeex of course the results may be more stable - i will try this